### PR TITLE
Support for GHC 7.10

### PR DESCRIPTION
**Pull request by jystic**
/cc @tranma

### CODE REVIEW

**Comment by tranma** [comment]
some nasty dependency errors! changes looks ok though :pig: when :green_apple:

File: Setup.hs [hunk 1]
```diff
@@ -50,12 +60,19 @@ genBuildInfo verbosity pkg info = do
...
+timestamp :: IO String
+timestamp =
+  readProcess "date" ["+%Y%m%d%H%M%S"] "" >>= \s ->
+    case splitAt 14 s of
+      (d, n : []) ->
+        if (length d == 14 && filter isDigit d == d)
+          then return d
+          else fail $ "date has failed to produce the correct format [" <> s <> "]."
+      _ ->
+        fail $ "date has failed to produce a date long enough [" <> s <> "]."
```
Comment (tranma) [inline Setup.hs]: what's wrong with using `formatTime`...? Is it slow?

**Comment by jystic** [comment]
yeah it's a bug where cabal/ghc will try to link  `Setup.hs` against all the versions of `unix` available

See https://www.reddit.com/r/haskell/comments/3634x2/cabal_is_giving_a_weird_error_when_attempting_to/

The fix is to not using anything other than base & Cabal in Setup.hs files, or wait for Cabal 1.24

File: Setup.hs (hunk 1 above)
Comment (jystic) [inline Setup.hs]: Similar problem to multiple `unix` packages, if you configure with 7.8.4 then try to configure with 7.10.2 it will break

Comment (tranma) [inline Setup.hs]: wow, so we can't even use `time` in `Setup.hs`. Savages

Comment (jystic) [inline Setup.hs]: nope, too risky for the moment

cabal-1.24 adds support for Setup.hs dependency lists

**Pull request merged**